pkg/storage/s3/storage.go

```diff
@@ -116,10 +116,6 @@ func (is *ObjectStorage) Unlock(lockStart *time.Time) {
 func (is *ObjectStorage) initRepo(name string) error {
 	repoDir := path.Join(is.rootDir, name)
 
-	if fi, err := is.store.Stat(context.Background(), repoDir); err == nil && fi.IsDir() {
```

Comment:
If you have a file a/b/c/d1/file and you use is.store.Stat(a/b/c/d) it will return that it exists and is a directory :/.

Reply:
this is an upstream issue

Reply:
```
Testing s3 lib bug
uploading /ab/cd/ef/gh/file.txt
Uploaded successfully
stat testFile, isDir: false, size: 17, path: /ab/cd/ef/gh/file.txt
stat /ab/cd/ef/gh, should be dir => isDir: true, size: 0, path: /ab/cd/ef/gh
stat /ab/cd/e, should not be dir => isDir: true, size: 0, path: /ab/cd/e
stat /a, should not be dir => isDir: true, size: 0, path: /a
```

Reply:
Now we can conclude that we can not rely on checking if fileInfo.isDir() is true because it has false positives but we can be sure that it doesn't have false negatives.

I've looked into our code and the only check for isDir is in initRepo which I already modified
the other one is in DirExists method which is not used so far.

I propose merging this PR to get rid of this error.

Reply:
Can we add a test which reproduces the initial error with the old code, and does not with this patch?

Reply:
Ok, sure.